src/lib/account-update.ts

```diff
@@ -1017,7 +1017,12 @@ class AccountUpdate implements Types.AccountUpdate {
     // implementations are equivalent, and catch regressions quickly
     if (Provable.inCheckedComputation()) {
       let input = Types.AccountUpdate.toInput(this);
-      return hashWithPrefix(prefixes.body, packToFields(input));
+      return hashWithPrefix(
+        activeInstance.getNetworkId() === 'mainnet'
+          ? prefixes.zkappBodyMainnet
+          : prefixes.zkappBodyTestnet,
+        packToFields(input)
+      );
     } else {
       let json = Types.AccountUpdate.toJSON(this);
       return Field(Test.hashFromJson.accountUpdate(JSON.stringify(json)));
```

Comment:
this will also need to take the network id

Reply:
Thanks @mitschabaude, can you please assist me where it comes from, the OCaml side I believe, isn't it?

Reply:
yes, from `consistency-test.ml`

Reply:
- [Test.hashFromJson.accountUpdate()](https://github.com/o1-labs/o1js/pull/1437/commits/1eb25efe8d980ab8784e7ff2d66957f3eeebe9cc) seems to be done.
- Next up `Test.hashFromJson.transactionCommitments()`.